mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,506 @@
+import numpy as np
+from mne.filter import band_pass_filter
+from mne.utils import _time_mask
+from mne.parallel import parallel_func
+from mne.time_frequency import cwt_morlet
+from mne.preprocessing import peak_finder
+from mne.utils import ProgressBar
+from mne.baseline import rescale
+
+
+# Supported PAC functions
+_pac_funcs = ['plv', 'glm', 'mi_tort', 'mi_canolty', 'ozkurt', 'otc']
+# Calculate the phase of the amplitude signal for these PAC funcs
+_hi_phase_funcs = ['plv']
+
+
+def phase_amplitude_coupling(inst, f_phase, f_amp, ixs, pac_func='ozkurt',
+                             ev=None, ev_grouping=None, tmin=None, tmax=None,
+                             baseline=None, baseline_kind='mean',
+                             scale_amp_func=None, use_times=None, npad='auto',
+                             return_data=False, concat_epochs=True, n_jobs=1,
+                             verbose=None):
+    """ Compute phase-amplitude coupling between pairs of signals using pacpy.
+
+    Parameters
+    ----------
+    inst : an instance of Raw or Epochs
+        The data used to calculate PAC
+    f_phase : array, dtype float, shape (2,)
+        The frequency range to use for low-frequency phase carrier.
+    f_amp : array, dtype float, shape (2,)
+        The frequency range to use for high-frequency amplitude modulation.
+    ixs : array-like, shape (n_pairs x 2)
+        The indices for low/high frequency channels. PAC will be estimated
+        between n_pairs of channels. Indices correspond to rows of `data`.
+    pac_func : string, ['plv', 'glm', 'mi_canolty', 'mi_tort', 'ozkurt']
+        The function for estimating PAC. Corresponds to functions in pacpy.pac
+    ev : array-like, shape (n_events,) | None
```

Comment:
(going to reply to all of these as a mental checklist, PR to come soon)

Good point - I tried to be a little more explicit in the parameter names this time (e.g., ev_ixs instead of ev, pac_times instead of use_times). Feel free to suggest better names.